sdk/lib/stellar/horizon_error.rb

```diff
@@ -0,0 +1,53 @@
+module Stellar
+  class HorizonError < StandardError
+    attr_reader :original_response, :extras
+
+    def self.make(hash)
+      klass = class_for(hash["title"])
+      klass.new(hash)
+    end
+
+    # Uses specialized class if it is defined
+    def self.class_for(title)
+      const_get title.delete(" ").to_sym
+    rescue NameError
+      self
+    end
+
+    def initialize(hash)
+      message = make_message(
+        detail: hash["detail"],
+        title: hash["title"]
+      )
+
+      # Superclass StandardError has different argument signature
+      super(message)
+
+      @original_response = hash
+      @extras = hash["extras"]
+    end
+
+    private
+
+    def make_message(detail:, title:)
+      # Include title only on generic class
+      # Non-equality is true only for child classes
+      if self.class != HorizonError
+        title = nil
+      end
+
+      [title, detail].join(": ")
```

Comment:
For non-generic class `title` will be `nil`, and this expression will result in `": detail"`, which is not what we would like to get 🙂